### DISCUSSION TITLE
Weekly Meeting

### DISCUSSION THREAD

**Discussion by bryevdv** [bokeh/bokeh#12568] [Governance]
Hi @bokeh/core 

First off, a personal note. I've be conducting the weekly meeting for most of the last ten years. I would like to take a break. :) I think that that would be good not only for me, but also for the value of the meeting itself. Is there anyone who is willing to take over conducting the meeting? Or perhaps a few people could rotate and share the responsibility? If any advice or mentoring would be useful, I am of course happy to provide. Mostly I think this is a great opportunity for others to develop their own ways of contributing in new areas (which as a result, will only *improve* our interactions).

Since the pandemic, I have shared a a home-office with my partner and have gotten to observe them have tight, efficient, orderly meetings. Along the theme of increasing the meeting value, I'd also like to offer suggestions for potential changes to how it is conducted, based on those learnings. 

# Suggestions

## Muting by default

Different people have different ways of engaging in conversation, according to language, culture, and background, especially around when and how to interject.  These differences can cause (unintentional) friction in conversations. (The fun term for this is [complimentary schizmogenesis](https://www.omniglot.com/bloggle/?p=86)). I personally think these issues are even more exacerbated in remote meetings, where we can't rely on body language.

This is something I struggle with; I *never* want to interrupt or overspeak anyone, but there are still times when my brain sincerely thought there was a good opening to chime in, but then it turns it *was not*. I can apologize in these instances, but what if we decide to use available tools to avoid these problems altogether? I would propose that in the future:

* Everyone be muted by default, unless they are the active speaker, and wait to be explicitly acknowledged before speaking.  To be acknowledged, use the meeting tools, e.g. "raise hand". 

## "Top N" updates

I've been in a few organizations that impose arbitrary caps on updates to help keep general sync meetings as useful to everyone as possible, while also ensuring everyone gets a space to contribute. These are usually in the form of "Top Five Things" or "T5T" rules. I'd propose that:

* Everyone should provide (at most) a "Top Five" most important updates for the week

Sometimes there *is* more to get into, which is *fine* and *expected*. But then in those cases, usually a smaller group is appropriate. One of the T5T can always be to discuss who should come together in a follow-up meeting to dive into anything else more deeply.

## Focus on impact 

We can all go look a the trackers and discussion forums any time to get a view of ongoing things there. The value of the meeting is highest (IMO) when it conveys information that makes a difference for other's work, that they might otherwise not know or be able to find out. I would suggest:

* Prioritize updates that: will (or have) direct effects for other team members, or for the community, are things you are blocked on/need assistance with, or are important things that don't otherwise have wide visibility among the group

## Actions

I tried to start recording actions from meetings, but that is only a good start. I am really bad about forgetting my own action items. I think we can promote more accountability (in an encouraging  way) with structured reminders about previous items. I would propose:

* Revisit and update the previous backlog od actions as the *first* agenda item each meeting. 

## Open agenda

Current the "agenda" is very spur of the moment, and I have generally just asked about things during the meeting. I think soliciting for agenda items during the meeting is fine, but it also puts folks on the spot. Other places I have been have a document where anyone can add agenda items *ahead* of the meeting.

* Create an explicit "agenda" document that participants can add discussion items to at their own convenience.

The document is just a scratch space, to refer to in the meeting to know what needs discussing, that anyone can easily update at any time. It could be a sticky discussion topic, wiki page, google doc, whatever is most convenient.

**Comment by ianthomas23** [comment]
Thanks for the write-up @bryevdv, you are very much entitled to take a step back from running the meetings.

I think it would be best if we divide meeting responsibilities into two:

- chair, who is responsible for starting and ending the meeting, prompting who can speak next, and meeting flow.
- note taker (or the more traditional term secretary), who is responsible for recording decisions made, actions, and so on.

I would be inclined to formalise the meeting process and responsibilities as much as possible, they only need to be a few bullet points each.  This should make the two responsibilities easier for people to volunteer for as the tasks are well-specified. I am happy to volunteer for either, or indeed both on an alternating basis.

What I would most like to change in the emphasis of the updates. They mostly tend to be "items completed" which are looking backwards at problems we had but are now solved. Many are closed issues and PRs which are already documented on github. I'd prefer the updates to be looking more forwards than backwards, so problems that are being worked on but yet to be solved, that may need discussion and perhaps even voting on. I suppose this is really just a paraphrase of the "Focus in impact" section above, but I would take it further and not talk about "updates" at all but only "agenda items" that individual add to the meeting document (ideally before but during the meeting would be OK too). Agenda items could still include work completed, if someone wanted to ensure that the ramifications of said work were fully understood by all present.

We should consider how the meeting notes will be recorded. The VSCode Live Share sometimes works seamlessly for me but other times does not. It might be that I just need more practice using it, but I do find hackmd easier to use. I wouldn't have a separate agenda document, but include this is in the meeting document, so each meeting has a section in the document divided up as follows:

- Date
- People present (with chair/note taker named)
- Agenda items (can be added before or during the meeting)
- Notes takes during meeting. Starts with check of last week's actions, then follows the previous section's agenda items in turn (as the meeting itself does). This week's actions could be with their associated notes, or highlighted at the start or end?

The agenda and notes sections are distinct so that anyone can append to the agenda items, but the note taker should be the only person writing or modifying the notes.

**Reply by bryevdv** [reply]
Thanks for your detailed thoughts @ianthomas23. Your framing as "forward-looking" is really very close to what I had in mind with "focus on impact", i.e. if we are going to invest the time to communicate things, they should be things that could impact future effort (by definition, looking forward). 

I also agree that having both a chair and a recorder would be the ideal situation. Speaking from experience, it is hard for me to do both at the same time (which is part of my impetus to take a break).

**Reply by bryevdv** [reply]
About tools: 

> The VSCode Live Share sometimes works seamlessly for me but other times does not. It might be that I just need more practice using it, but I do find hackmd easier to use.

I think there is room for experimentation. FWIW I switched from hackmd to vscode because it was much easier for *me* (HackMD UX was very complicated and confusing and seemed at odd with simple git notions sometimes, like I just want to commit and push and somehow that was 10 UI interactions... maybe it is better now).

**Comment by pavithraes** [comment]
Thank you, @bryevdv and @ianthomas23 for sharing your thoughts and ideas here! We also discussed this during the PyData Global  sprint and on Slack, here is the concrete proposal for weekly sync meetings starting Jan 2023: https://hackmd.io/@bokeh/B10wdzFwi

HackMD isn't perfect, but it'll allow more folks to collaborate on the notes compared to VSCode liveshare, especially non-developers. :)

**Reply by mattpap** [reply]
> Every Wednesday, at 8am Pacific Time

Would have been better if this was universal time.

**Comment by ianthomas23** [comment]
Here are some explanatory notes about the proposal:

1. It is quite structured. This makes it easy for the moderator and notetaker to follow the process: start at the top, work down to the bottom.
2. There is no explicit round table of updates. If you have something to say append it to the discussion items or external announcements, ideally before the meeting but during the meeting will be OK too.
3. We have kept in a section for updates which is specifically for those who are doing open funded work to summarise their weekly progress. Currently this is for Mateusz's Anaconda-funded work and CZI round 5 which starts shortly.
4. The fundamental idea is that the meeting flows from the past (past actions and updates) through to the present (discussion items about things that matter today) and then the future (announcements and future actions).
5. Moderator and notetaker will rotate responsibilities. There will inevitably be variations in how the roles are approached and we can evolve best practice by learning from each other's ideas. More on this later.
6. Once we've tried it for a few weeks we can check if it is working, what we need to change, etc.

**Comment by ianthomas23** [comment]
Initially we have myself, Pavithra and Timo to take it in turns to be moderator and notetaker:

- Week 1, January 4th: moderator = Ian, notetaker = Pavithra
- Week 2, January 11th: moderator = Pavithra, notetaker = Timo
- Week 3, January 18th: moderator = Timo, notetaker = Ian

and so on. If the moderator/notetaker cannot make a particular meeting the spare person can stand in for them.

We'll put the names of the moderator and notetaker on the meeting notes before the meeting so that everyone knows who is doing what.
